### PR TITLE
[4.0] Fix modal not opening in subfield

### PR DESCRIPTION
**Pull request by Quy**
Pull Request for Issue #26711 and #28953.

Thanks @dgrammatiko for the code.

### Testing Instructions
Install patch and `npm run build:js`
or 
Download installer package.

Create 2 custom fields: Media and User
Create a subfield consisting these 2 custom fields.
Edit an article.
Go to Fields tab.
Add a row.
Click Select button.
Click User icon.

### Actual result BEFORE applying this Pull Request
Modal doesn't open.

### Expected result AFTER applying this Pull Request
Modal opens.

### CODE REVIEW

**Comment by HRIT-Florian** [comment]
I have tested this item :white_check_mark: successfully on aacd7ea9adb8214d7d8b928738c66b872715922e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32624">issues.joomla.org/tracker/joomla-cms/32624</a>.</sub>

Thx

File: build/media_source/system/js/fields/joomla-field-media.w-c.es6.js [hunk 1]
```diff
@@ -100,7 +100,7 @@
       if (this.modalElement
         && window.bootstrap
         && window.bootstrap.Modal
-        && window.bootstrap.Modal.getInstance(this.modalElement) === undefined) {
+        && window.bootstrap.Modal.getInstance(this.modalElement) === null) {
```
Comment (Fedik) [inline build/media_source/system/js/fields/joomla-field-media.w-c.es6.js]: better do:
 `&& !window.bootstrap.Modal.getInstance(this.modalElement)) {`

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on aacd7ea9adb8214d7d8b928738c66b872715922e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32624">issues.joomla.org/tracker/joomla-cms/32624</a>.</sub>

**Comment by dgrammatiko** [comment]
@Quy please do the same change in the user modal: https://github.com/joomla/joomla-cms/blob/adce349e48a0bd53c7ca4bf5bd327e1b5876d657/build/media_source/system/js/fields/joomla-field-user.w-c.es6.js#L60

File: build/media_source/system/js/fields/joomla-field-media.w-c.es6.js (hunk 1 above)
Comment (Quy) [reply build/media_source/system/js/fields/joomla-field-media.w-c.es6.js]: Thanks!

**Comment by brianteeman** [comment]
This resolves the issue for media and user fields but the problem still exists for other modals eg article select

**Comment by dgrammatiko** [comment]
> This resolves the issue for media and user fields but the problem still exists for other modals eg article select

What's the problem with them? This is an issue due to the way subforms work and the fact that some fields have still dependency on Bootstrap modal. The other modals are tightly connected only to editors...

**Comment by brianteeman** [comment]
there is no content in the modal. Just the header and footer

**Comment by dgrammatiko** [comment]
URL?

**Comment by brianteeman** [comment]
will send it privately

**Comment by brianteeman** [comment]
scrub - that. I must have had a typo or something. Its working now

**Comment by brianteeman** [comment]
I was wrong - its not working but it is unrelated - sorry for the confusion on this PR

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on f49c6977a22c058f014140d6131a10b48856623a<br /><br />Tested successfully with media and user fields<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32624">issues.joomla.org/tracker/joomla-cms/32624</a>.</sub>

**Comment by brianteeman** [comment]
However it does NOT solve the issue for other modal fields such as administrator/components/com_content/src/Field/Modal/ArticleField.php

I'm guessing there needs to be a patch for `modal-field.js`

**Comment by sksuryan** [comment]
I have tested this item :white_check_mark: successfully on f49c6977a22c058f014140d6131a10b48856623a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32624">issues.joomla.org/tracker/joomla-cms/32624</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32624">issues.joomla.org/tracker/joomla-cms/32624</a>.</sub>

**Comment by chmst** [comment]
Thanks!

**Pull request merged**